cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java

```diff
@@ -216,6 +232,80 @@ private void addInstancesNotKnownByCMWithLimit(Integer limit, Set<InstanceMetaDa
         }
     }
 
+    public void enterMaintenanceMode(Stack stack, Set<String> hostList, ApiClient client) {
+        HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(client);
+        String currentHostId = null;
+        int successCount = 0;
+        List<String> availableHostsIdsFromCm = null;
+        LOGGER.debug("Attempting to put {} instances into CM maintenance mode", hostList == null ? 0 : hostList.size());
+        try {
+            ApiHostList hostRefList = hostsResourceApi.readHosts(null, null, SUMMARY_REQUEST_VIEW);
+            availableHostsIdsFromCm = hostRefList.getItems().stream()
+                    .filter(apiHostRef -> hostList.contains(apiHostRef.getHostname()))
+                    .parallel()
+                    .map(ApiHost::getHostId)
+                    .collect(Collectors.toList());
+
+            for (String hostId : availableHostsIdsFromCm) {
+                currentHostId = hostId;
+                hostsResourceApi.enterMaintenanceMode(hostId);
```

Comment:
I can see async method, maybe it would be better to use it (in high node count case it can take long time)

Reply:
also worth to request a batch API for this from CM team in the future

Reply:
This is a fairly fast call. I'm not going near an async API for this, unless it is patch. Also didn't ask for a batch API because it is already reasonably fast.

Reply:
I am fine with it if it is very fast (1-2 sec for 1-200 nodes).

Reply:
even if it is fast, puts high load on network traffic (thus on CCM/cluster proxy) and possibly it can fail easily if there is a network issue during the time interval of these calls

Reply:
not to mention if let's say the customer is in Europe and using us-west CDP and we need to wait for the round-trip for every node